### DISCUSSION TITLE
App crashes if you insert an imported API_KEY from .env instead of writing it directly

### DISCUSSION THREAD

**Discussion by yaroslav-vasilyev** [react-native-maps/react-native-maps#4886] [Q&A]
Hi guys)
Since it is considered that storing things like API_KEY in native code is not safe from a cyber security point of view, so I am addressing you with this question. I have this function
```
static void implementGoogleMaps() {
  NSString *MAP_API_KEY = [ReactNativeConfig envFor:@"MAP_API_KEY"];
  if (MAP_API_KEY) {
    [GMSServices provideAPIKey:[ReactNativeConfig envFor:@"MAP_API_KEY"]];
  } else {
    NSLog(@"Missing MAP_API_KEY in .env").
  }
}
```

Which should essentially work without error, since I just import the variable from .env before calling the necessary `GMSServices provideAPIKey` function. However, I'm encountering that the application crashes after I get to the `MapView` component page. Perhaps you know how to solve this problem. And maybe you have solved the API_KEY security issue in some other way, so that you don't have to write it directly in native code? I will be grateful for any answers!

**Comment by 0xSuryax** [comment]
[TRY THIS AND RESTRICT YOUR KEY FROM GOOGLE CLOUD CONSOLE WITH APPLICATION SIGNATURE](https://github.com/react-native-maps/react-native-maps/commit/974efa50767ff3016b8c09c6fb5ad40f2cd63212)
